grai-server/app/the_guide/settings.py

```diff
@@ -14,21 +15,55 @@
 # SECURITY WARNING: keep the secret key used in production secret!
 
 
-def clean_allowed_hosts(val):
+def clean_hosts(val):
     if isinstance(val, list):
         return [item.strip() for item in val]
     elif isinstance(val, str):
         return [s.strip() for s in val.split(",")]
     else:
-        raise
+        raise TypeError(f"hosts must be a list or a string not {type(val)}")
 
 
 SECRET_KEY = config("SECRET_KEY", default=get_random_secret_key())
-DEBUG = config("DEBUG", default=True, cast=bool)
+DEBUG = config("DEBUG", default=False, cast=bool)
 TEMPLATE_DEBUG = config("TEMPLATE_DEBUG", default=DEBUG, cast=bool)
-ALLOWED_HOSTS = config(
-    "ALLOWED_HOSTS", default=".localhost, 127.0.0.1, [::1]", cast=clean_allowed_hosts
-)
+
+SERVER_HOST = config("SERVER_HOST", default="localhost", cast=str)
+SERVER_PORT = config("SERVER_PORT", default="8000", cast=str)
+
+FRONTEND_HOST = config("FRONTEND_HOST", default=SERVER_HOST, cast=str)
+FRONTEND_PORT = config("FRONTEND_PORT", default="3000", cast=str)
+
+ALLOWED_HOSTS = config("ALLOWED_HOSTS", default=[], cast=clean_hosts)
+CORS_ALLOW_ALL_ORIGINS = config("CORS_ALLOW_ALL_ORIGINS", default=False, cast=bool)
+CORS_ALLOWED_ORIGINS = config("CORS_ALLOWED_ORIGINS", default=[], cast=clean_hosts)
+CSRF_TRUSTED_ORIGINS = config("CSRF_TRUSTED_ORIGINS", default=[], cast=clean_hosts)
+
+if DEBUG:
+    ALLOWED_HOSTS = ["*"]
+    CORS_ALLOW_ALL_ORIGINS = True
+    CSRF_TRUSTED_ORIGINS = ["http://*", "https://*"]
+else:
+    schemes = ["http", "https"]
+    default_ports = [SERVER_PORT, FRONTEND_PORT]
+    default_hosts = list(set([SERVER_HOST, FRONTEND_HOST]))
+    default_cors_origins = [
+        f"{scheme}://{host}:{port}"
+        for scheme, host, port in product(schemes, default_hosts, default_ports)
+    ]
```

Comment:
Is this crossing all three of these arrays?  I guess it works

Reply:
Yes, exactly. It's producing every combination of scheme, host, and port.